Add Pull Request Checklist (#1520)
TheThingsNetwork/lorawan-stack: .github/PULL_REQUEST_TEMPLATE.md

-->

...

#### Checklist
<!-- Make sure that this pull request is complete. -->

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.

> - [ ] Compatibility: The changes do not break existing API, database or configuration.

Reviewer: ```suggestion
- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration.
```

Reviewer 2: I think compatibility should also cover migrations:
`The changes are backwards compatible with existing API, database and configuration. In cases where the database schema has been changed, the migrations process should work correctly.`.

Reviewer 3: This cannot and should not be exhaustive here as a checkbox. We have a compatibility commitment, just link to that, by checking that it aligns with the compatibility commitment.

> - [ ] Testing: The changes are covered with unit tests. The changes have been tested by hand.

Reviewer: ```suggestion
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
```

- [ ] Documentation: Relevant documentation is added or updated.

> - [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to the changelog.

Reviewer 3: Add link

> - [ ] Commits: Commit messages follow pattern, no fixup commits left.

Reviewer 3: Add link
